src/middlewared/middlewared/test/integration/assets/smb.py

```diff
@@ -23,3 +32,21 @@ def smb_share(path, name, options=None):
     finally:
         call("sharing.smb.delete", share["id"])
         call("service.stop", "cifs")
+
+
+@contextlib.contextmanager
+def smb_mount(share, username, password, local_path='/mnt/cifs', options=None, ip=default_ip):
+    mount_options = [f'username={username}', f'password={password}'] + (options or [])
+    mount_cmd = [
+        'mount.cifs', f'//{ip}/{share}', local_path,
+        '-o', ','.join(mount_options)
+    ]
+
+    mount_string = ' '.join(mount_cmd)
+
+    ssh(f'mkdir {local_path}; {mount_string}')
```

Comment:
We should use `shlex.quote` in case `local_path` contains spaces. Or remove the ability to redefine it as it is not redefined by any caller.

Reply:
There will be more mounts used as I add more tests.